Inttest

lugerard opened the pull request:
This adds integration test using VSCode test electron, Mocha and Sinon to qt-core, qt-qml, qt-ui an qt-cpp.
For qt-core, part of the tests for the extension commands are present.

<!---
# Qt contribution guidelines

We welcome contributions to Qt!

Read the
[Qt Contribution Guidelines](https://wiki.qt.io/Qt_Contribution_Guidelines) to learn more.
<!---

lugerard committed:
qt-cpp: Add installing extension depedencies in test Host

lugerard committed:
qt-cpp: Add VS Code integration test

lugerard committed:
qt-cpp: Empty test working

lugerard committed:
qt-cpp: Continue adding VS Code integration test

Build-test pointing to tsconfig.test.json
Exclude there are ignored
All out files are regenerated when building the test
this is because the test imports the path

lugerard committed:
qt-cpp: Test built in esbuild.js

lugerard committed:
qt-cpp: Introducing compile-test

Build test in esbuld.mjs in a modular manner
Call compile-test in the task called when launching
the test.

lugerard committed:
qt-cpp: Modify test script to use compile-test

lugerard committed:
qt-cpp: Change test files glob path

lugerard committed:
qt-cpp: Add testing cmake extension is installed

Calling a cmake command

lugerard committed:
qt-cpp: Add testing qtcore extension is installed

Doing this by checking that
vscode.commands.getCommands sees the qt-core
command.

The qt-core extension needs to be activated first

There might be a nicer wait to check that the
extension is installed. Simply checking if it is active?

lugerard committed:
qt-cpp: Add checking extension depedencies activation

Removes cmake and qt-core previous individual tests

lugerard committed:
qt-cpp: Add testing qt-core commands visibility

lugerard committed:
qt-cpp: Add license and copyright to test files

Small clean up

lugerard committed:
qt-cpp: Test switch to bdd

Needs cleaning
Simple stub with dummy function working
Test example shows that stub can't reach other extension.
Stub is used if stubbed function is called from qt-cpp
but not if called from qt-core

lugerard committed:
qt-cpp: Use local qt-core extension in test

Needs cleaning
Add some examples on how to set things in qt-core from qt-cpp test
Via:
- command
- api
- properties

lugerard committed:
qt-cpp: Cleaning test

- Test structure and basic test in place

lugerard committed:
qt-core: Add integration test

Integration test structure is added at qt-core level
Only basic test are implemented.

lugerard committed:
qt-qml: Add integration test

- Test structure and basic test in place

lugerard committed:
qt-ui: Add integration test

- Test structure and basic test in place

lugerard committed:
qt-core: Add test for commands (Part1/2)

lugerard committed:
Set helper for extensions test in qt-lib

OrkunTokdemir commented on qt-core/src/test/suite/extension.test.ts:
```diff
@@ -0,0 +1,30 @@
+describe('extension', () => {
+  before('activate', async function () {
+    const ext = vscode.extensions.getExtension('theqtcompany.qt-core');
+    if (!ext) throw new Error('qt-qml extension not found');
```
`qt-core extension not found`?

lugerard replied:
done

OrkunTokdemir commented on .vscode/launch.json:
```diff
@@ -13,6 +13,19 @@
+      "request": "launch",
+      "runtimeExecutable": "${execPath}",
+      "args": [
+              //"--disable-extensions",
```
Forgot to delete?
Repeats.

lugerard replied:
done

OrkunTokdemir commented on qt-core/src/installation-root.ts:
```diff
@@ -163,7 +163,7 @@ export async function registerQt() {
   const options: vscode.OpenDialogOptions = {
     canSelectMany: false,
-    openLabel: 'Select Qt installation root',
+    openLabel: 'Select Qt installation root please',
```
Do we need this change?

lugerard replied:
no

OrkunTokdemir commented on qt-core/src/extension.ts:
```diff
@@ -82,7 +92,7 @@ export async function activate(context: vscode.ExtensionContext) {
   checkQtpathsInEnvPath();
   initCoreValues();
-  return coreAPI;
+  return { coreAPI, projectManager };
```
I think we should not return the project manager here, or at least in production.

lugerard replied:
removed

OrkunTokdemir commented on qt-core/src/test/helper.ts:
```diff
@@ -0,0 +1,739 @@
+import { expect } from 'chai';
+import * as sinon from 'sinon';
+import * as vscode from 'vscode';
+import { QtInfo, CoreAPI } from 'qt-lib';
```
Can you separate internal and external imports with a space?

OrkunTokdemir commented on qt-core/src/test/suite/commands.test.ts:
```diff
@@ -0,0 +1,666 @@
+  generateDefaultQtPathsName
+} from 'qt-lib';
+import * as path from 'path';
+//import isEqual from 'lodash/isEqual';
```
Forgot to delete?
Repeats

OrkunTokdemir commented on qt-core/src/test/helper.ts:
```diff
@@ -0,0 +1,739 @@
+  sb: sinon.SinonSandbox,
+  matchingOptions: vscode.OpenDialogOptions,
+  urisToReturn: NonEmptyArray<vscode.Uri> | undefined // Only user interaction possibilities
+): sinon.SinonSpy {
```
NIT: We generally don't write return types.
Repeats

OrkunTokdemir commented on .vscode/tasks.json:
```diff
@@ -96,6 +147,22 @@
         "cwd": "${workspaceFolder}"
       }
+    },
+    {
```
```suggestion
    {
    "type": "shell",
```
This is missing here

lugerard replied:
I don't think this is coming from my changes. Do we add it in this pull request anyway?

lugerard replied:
done

OrkunTokdemir commented on qt-cpp/src/commands/scan-qt-kits.ts:
```diff
@@ -2,12 +2,13 @@
 import { EXTENSION_ID } from '@/constants';
 
 export function registerScanForQtKitsCommand() {
+  void vscode.window.showInformationMessage('Testing Window');
```
Can you please delete this?

lugerard replied:
done

OrkunTokdemir commented on qt-cpp/src/test/suite/index.ts:
```diff
@@ -0,0 +1,56 @@
+export function run(): Promise<void> {
+  // Create the mocha test
+  const mocha = new Mocha({
+    // suite, test
```
Forgot to delete?
Repeats

lugerard replied:
done

OrkunTokdemir commented on package.json:
```diff
@@ -12,6 +12,7 @@
     "all_dev:qt-core": "ts-node ./scripts/all_dev.ts --extension=\"qt-core\"",
     "all_dev:qt-qml": "ts-node ./scripts/all_dev.ts --extension=\"qt-qml\"",
     "all_dev:qt-cpp": "ts-node ./scripts/all_dev.ts --extension=\"qt-cpp\"",
+    "all_dev_test": "npm run all_dev:qt-cpp && npm run compile-test",
```
This one is not complete. Can you fix and also add commands per extension, like the rest of the commands?

lugerard replied:
done

OrkunTokdemir reviewed (changes requested):
Generally, it looks good, and it will be good infrastructure for the future for us but there are small things that need to be changed before merging it. I shared my comments on those.

OrkunTokdemir reviewed (changes requested):
Can you please also move the `test` folder outside of `src` so that we can separate them completely?

lugerard committed:
qt-core: adapt to node16

lugerard commented:
Will do.


Confidential

________________________________
From: Orkun Tokdemir ***@***.***>
Sent: Monday, August 18, 2025 1:59 PM
To: qt-labs/vscodeext ***@***.***>
Cc: Lucie Gérard ***@***.***>; Author ***@***.***>
Subject: Re: [qt-labs/vscodeext] Inttest (PR #188)


@OrkunTokdemir requested changes on this pull request.

Can you please also move the test folder outside of src so that we can separate them completely?

—
Reply to this email directly, view it on GitHub<https://github.com/qt-labs/vscodeext/pull/188#pullrequestreview-3128078252>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/API3OMR5PDDLS2U5O45LMUL3OG5ZZAVCNFSM6AAAAACCCP7GL6VHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZTCMRYGA3TQMRVGI>.
You are receiving this because you authored the thread.Message ID: ***@***.***>

lugerard committed:
qt-core: remove returning CoreManager and use in test

lugerard committed:
Updating ThirdPartyNotices

lugerard committed:
Cleaning

lugerard committed:
qt-core: Move test out of src/

lugerard committed:
qt-cpp: Move test out of src/

lugerard committed:
qt-qml: Move test out of src/

lugerard committed:
qt-ui: Move test out of src/

lugerard committed:
qt-core: Add test for commands (Part2/2)

Add test env logic with globally stored factories.
This allows to take control and/or spy on extension logic
when unavoidable without having to stub the extension.

lugerard committed:
qt-core: Simplify createNewItem test

Remove factory provider and usage.
Delay instead.

temp2: working with sleep
